rest/class-boldgrid-backup-rest-archive.php

```diff
@@ -0,0 +1,277 @@
+<?php
+/**
+ * File: class-boldgrid-backup-rest-archive.php
+ *
+ * @link       https://www.boldgrid.com
+ * @since      X.X.X
+ *
+ * @package    Boldgrid_Backup
+ * @copyright  BoldGrid
+ * @version    $Id$
+ * @author     BoldGrid <support@boldgrid.com>
+ */
+
+/**
+ * Class: Boldgrid_Backup_Rest_Archive
+ *
+ * REST endpoints to perform simple archive manipulation.
+ *
+ * @since X.X.X
+ */
+class Boldgrid_Backup_Rest_Archive extends Boldgrid_Backup_Rest_Controller {
+
+	/**
+	 * Resource name.
+	 *
+	 * @since  X.X.X
+	 * @access private
+	 * @var    string
+	 */
+	protected $resource = 'archives';
+
+	/**
+	 * Register all routes.
+	 *
+	 * @since X.X.X
+	 */
+	public function register_routes() {
+		$this->register_creation();
+		$this->register_restore();
+		$this->register_list();
+	}
+
+	/**
+	 * Register the route for creating an archive.
+	 *
+	 * @since X.X.X
+	 */
+	public function register_creation() {
+		register_rest_route( $this->namespace, '/' . $this->resource, [
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'create_item' ],
+				'permission_callback' => [ $this, 'permission_check' ],
+			],
+			'schema' => [ $this, 'get_schema' ],
+		] );
+	}
+
+	/**
+	 * Register the route for restoring a backup.
+	 *
+	 * @since X.X.X
+	 */
+	public function register_restore() {
+		register_rest_route( $this->namespace, '/' . $this->resource, [
+			[
+				'methods'             => 'PUT',
+				'callback'            => [ $this, 'restore' ],
+				'permission_callback' => [ $this, 'permission_check' ],
+				'args'                => [
+					'url' => [
+						'required'            => false,
+						'description'         => esc_html__( 'Route URL to restore.', 'boldgrid-backup' ),
+						'type'                => 'string',
+						'sanitation_callback' => function ( $field ) {
+							return esc_url_raw( $field );
+						},
+					],
+					'id'  => [
+						'required'            => false,
+						'description'         => esc_html__( 'Backup id to restore.', 'boldgrid-backup' ),
+						'type'                => 'int',
+						'sanitation_callback' => function ( $field ) {
+							return (int) $field;
+						},
+					],
+				],
+			],
+			'schema' => [ $this, 'get_schema' ],
+		] );
+	}
+
+	/**
+	 * Register the route for viewing a list of backups.
+	 *
+	 * @since X.X.X
+	 */
+	public function register_list() {
+		register_rest_route( $this->namespace, '/' . $this->resource, [
+			[
```

Comment:
Arrays should be declared with long array syntax per WordPress coding standards
https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays

Reply:
See similar comment above.